File: step07_appsync_with_subscriptions_using_amplify/aws-cdk/lambda/addTodo.ts
import { randomBytes } from 'crypto';
import { TodoInput } from './Todo';
const AWS = require('aws-sdk');
const docClient = new AWS.DynamoDB.DocumentClient();

async function addTodo(todo: TodoInput) {
    const params = {
        TableName: process.env.TODOS_TABLE,
        Item: { id: randomBytes(16).toString("hex"), ...todo }
    }
    try {
        await docClient.put(params).promise();
        return { ...params.Item };
    } catch (err) {
        console.log('DynamoDB error: ', err);
        return null;
    }
}

export default addTodo;

File: step07_appsync_with_subscriptions_using_amplify/aws-cdk/python/lambda/addTodo.py
import boto3
import os
import logging
from botocore.exceptions import ClientError

dynamodb = boto3.resource('dynamodb')
tableName = os.environ['TODOS_TABLE']
table = dynamodb.Table(tableName)
print(tableName)


def addTodoItem(todo):
    try:
        res =table.put_item(
            Item=todo
        )

        return res
    except ClientError as e:
        logging.error(e) 